### PR TITLE
[Fix-10730][UI] fix bug where the environment name not works well

### LINKED ISSUE #10730: [Bug] [UI] Workflow Definition. default `environment name` showed wrong when changing `worker group`
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

in the workflow definition. the default `environment name` showed wrong when changing `worker group`
wrong-env

### What you expected to happen

default `worker group` correctly changed when  `environment name` changed
expect

### How to reproduce

1. drag a node in the workflow definition
2. change worker group

### Anything else

it happened when 2 worker groups have the same length environment options.

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by rockfang**
## Purpose of the pull request
close #10730 

## Brief change log
when workerGroup changes, change the default environment name.

 works both in create and edit the dag node.
## Verify this pull request
This pull request is code cleanup without any test coverage.

### CODE REVIEW

**Comment by rockfang** [comment]
here are some screenshots.
1. create mode and only one in options. make the one as default value.
create-expect-one
2. create mode and more than one in options. no default value

create-expected
3. edit mode
edit-modal

File: dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-environment-name.ts [hunk 1]
```diff
@@ -54,23 +53,20 @@ export function useEnvironmentName(
...
       options.value = environmentList.filter((option: IEnvironmentNameOption) =>
         filterByWorkerGroup(option)
       )
+      // default environmentCode changing when workerGroup changed.
+      // if options only one. set the one as default.
+      if (options.value.length === 1) {
+        model.environmentCode = options.value[0].value
+      } else {
+        model.environmentCode = null
+      }
```
Comment (Amy0104) [inline dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-environment-name.ts]: Data playback will get a problem. Such as you have two workerGroups A and B. And then set the workerGroup to B and set the environment to B-2.  Save the task, and reopen you will get the environment to B-1.

**Review by Amy0104 — CHANGES_REQUESTED** [review]

File: dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-environment-name.ts (hunk 1 above)
Comment (rockfang) [reply dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-environment-name.ts]: got it. I will try to fix it later.

File: dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-worker-group.ts [hunk 2]
```diff
@@ -43,7 +42,8 @@ export function useWorkerGroup(): IJsonItem {
     span: 12,
     name: t('project.node.worker_group'),
     props: {
-      loading: loading
+      loading: loading,
+      'on-update:value': () => (model.environmentCode = null)
```
Comment (Amy0104) [inline dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-worker-group.ts]: Setting to null, the environmentCode will not has default value after changing workgroup.

**Review by Amy0104 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**